### DISCUSSION TITLE
Best way to get a direct stream from Frigate (http not RTSP)

### DISCUSSION THREAD

**Discussion by thehijacker** [blakeblackshear/frigate#5201] [Ask A Question] [answered]
Hello,

Ditched Blue Iris and now using only Frigate. Still needs some fine tuning with zones to avoid false positives and detection of stationary cars, but I am slowy getting there.

In Blue Iris I was able to get the MJPEG stream out of it to display in web pages (to pass to my tinyCam on phone, to display in web page and pass the web page to Google Hub display, ...). But in Frigate all I can get using restream is RTSP stream. What would be best way to get MJPEG or something simillar out of Frigate? Something I can display in web page and is compatible with all browsers and I can cast to Goole Hub display speaker.

I am also using Home Assistant integration. The FPS in camera video that I see is 2-5 fps. Very slow. With RTSP stream (rtsp://frigateIP:8554/cameraname) I can see it in full FPS. But in Home Assistant it is a HTTP stream that I can use using HA token and NGINX proxy to it. This is what I am using now to display on web page. Kind of step back that I need HA to see my camera feeds.

Thank you for all the tips.

**Comment by NickM-27** [comment, marked as answer]
All of these things you're mentioning can be accomplished with https://github.com/AlexxIT/WebRTC

If you truly need mjpeg then I'd suggest reading the docs https://docs.frigate.video/integrations/api#get-apicamera_name

**Reply by thehijacker** [reply]
Amazing. MJPEG stream over API works great. I use it from time to time anyway. Will also check the WebRTC integration. Direct casting looks promising. Now using workarounds. Thank you!

Just one more question. FPS in MJPEG stream is still 2 fps. No matter what parameter I set for ?fps=XX. Where is this lowered from original camera FPS? Watching line RTSP I get full FPS.

**Reply by thehijacker** [reply]
Found it. Live uses detect settings. Default is 5 fps. I changed it to 10 fps. Much better with slight CPU increase.

![image](https://user-images.githubusercontent.com/13857043/214061615-03a47723-6d9e-4aaf-a1ba-6b20e39fa790.png)
